authz/src/main/java/io/grpc/authz/AuthorizationPolicyTranslator.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.authz;
+
+import com.google.common.collect.ImmutableList;
+import io.envoyproxy.envoy.config.rbac.v3.Permission;
+import io.envoyproxy.envoy.config.rbac.v3.Policy;
+import io.envoyproxy.envoy.config.rbac.v3.Principal;
+import io.envoyproxy.envoy.config.rbac.v3.Principal.Authenticated;
+import io.envoyproxy.envoy.config.rbac.v3.RBAC;
+import io.envoyproxy.envoy.config.rbac.v3.RBAC.Action;
+import io.envoyproxy.envoy.config.route.v3.HeaderMatcher;
+import io.envoyproxy.envoy.type.matcher.v3.PathMatcher;
+import io.envoyproxy.envoy.type.matcher.v3.RegexMatcher;
+import io.envoyproxy.envoy.type.matcher.v3.StringMatcher;
+import io.grpc.internal.JsonParser;
+import io.grpc.internal.JsonUtil;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Translates a gRPC SDK authorization policy in JSON string to Envoy RBAC policies.
+ */
+public class AuthorizationPolicyTranslator {
```

Comment:
This shouldn't be public, right?

Reply:
I am getting the following error due to absence of public API in package.
javadoc: error - No public or protected classes found to document. Working on the fix.

Reply:
We can just disable javadoc or tell it to be a warning instead. I know we've fought that before, but I don't remember offhand how we solved it.

Disabling javadoc would be something akin to `javadoc.enable = false`. Making it a warning [looks to be](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.javadoc.Javadoc.html) `javadoc.failOnError = false`.

Reply:
I excluded authz directory from javadoc. Let me know if this will do. Thank you.